### PR TITLE
[transactions] Implement hifi Transactions screen and Transaction Details screen

### PR DESCRIPTION
**Pull request by selene-huang**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Updated Transactions screen to match hifi designs
- Added functionality to click on the entire transaction row instead of only the icon
- Added a Transaction Details screen
- Connected to Firebase so that both screens display actual data instead of dummy data
- Updated navigation functionality to allow for navigating between Transactions and Transaction Details
- Added a `getVendorByEmail` query to `src/database/queries.ts`
- Added functionality to retrieve transactions based on the current user's uuid
   - Refactored authContext and related functions to account for this new functionality
- Added loading spinner while fetching transactions and vouchers to Transactions screen and Transaction Details screen
- Added common utility functions under `src/utils/displayUtils.ts` (`formatValueForDisplay` and `formatTimeForDisplay`)

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
image
image

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
Run `npx expo start` and navigation to the Transactions screen. Verify that the transactions are scrollable and are displaying properly.

To check the Transaction Details screen, it's important that you click on the transaction labeled $10.12. Any other transaction will error due to recent schema changes.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Filtering will be added in future PRs

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"
Ran into a bug with an "excessive number of pending callbacks". This was resolved by adding an additional dependency to the useEffect.

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
Updated midfi code from #13 

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: src/components/Transactions/TransactionCard.tsx [hunk 1]
```diff
@@ -1,45 +1,69 @@
...
+      }}
+    >
+      <Row>
+        <DateContainer>
+          <Body1Text>{time.format('M/D')}</Body1Text>
+          <Body1Text>{time.format('h:mmA')}</Body1Text>
+        </DateContainer>
+
+        <ValueContainer>
+          <H3Subheading>${(value / 100).toFixed(2)}</H3Subheading>
```
Comment (kbrgl) [inline src/components/Transactions/TransactionCard.tsx]: Could be good to factor out the `${(value / 100).toFixed(2)}` stuff into a separate function called `formatCentsForDisplay` or something, seems like it's used in multiple places. Add this function in a new file under the `src/utils/` folder

File: src/components/Transactions/VoucherCard.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+type VoucherCardProps = {
+  serialNumber: number;
+  value: number;
+};
+export default function VoucherCard({ serialNumber, value }: VoucherCardProps) {
+  return (
+    <Row>
+      <DateContainer>
+        <Body1SemiboldText>${(value / 100).toFixed(2)}</Body1SemiboldText>
```
Comment (kbrgl) [inline src/components/Transactions/VoucherCard.tsx]: Use the `formatCentsForDisplay` function I mentioned in the other comment

File: src/components/Transactions/styles.ts [hunk 3]
```diff
@@ -8,38 +9,82 @@ export const Row = styled.View`
...
+
+export const RedText = styled.Text`
+  color: ${Colors.alertRed};
+  font-family: 'manrope-bold';
+  font-style: normal;
+  font-size: 12px;
+  line-height: 14px;
 `;
 
 export const Styles = StyleSheet.create({
```
Comment (kbrgl) [inline src/components/Transactions/styles.ts:101]: Strictly speaking, `Styles` should instead be lowercase—`styles`. Capitalized first letter is convention for components (which all the styled component definitions above implicitly are).

But perhaps that's something to change in a different PR, since you didn't add this in this PR? Ask Annie :)

File: src/screens/Transactions/TransactionDetailsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                <VoucherCard
+                  serialNumber={item.serialNumber}
+                  value={item.value}
+                />
+              )}
+              keyExtractor={item => item.serialNumber.toString()}
+            />
+          </CardContainer>
+        </TransactionsContainer>
+      ) : null}
```
Comment (kbrgl) [inline src/screens/Transactions/TransactionDetailsScreen.tsx]: Instead of showing nothing when the data hasn't been loaded yet, show a spinner?

File: src/screens/Transactions/styles.ts [hunk 5]
```diff
@@ -1,32 +1,79 @@
...
+
+export const MediumText = styled.Text`
+  font-family: 'manrope-regular';
+  font-style: normal;
+  font-size: 16px;
+  line-height: 22px;
+  text-align: center;
+`;
+
+export const Body1BoldText = styled.Text`
```
Comment (kbrgl) [inline src/screens/Transactions/styles.ts]: Use a more descriptive name, I know there's some notion of `Body1Text` in this codebase but that needs to be changed so may as well get a headstart

File: src/screens/Transactions/styles.ts [hunk 6]
```diff
@@ -1,32 +1,79 @@
...
 `;
 
-export const Title = styled.View`
-  padding-top: 20px;
-  padding-bottom: 20px;
+export const StartOfListView = styled.View`
+  width: 100%;
+  height: 1px;
+  border: 0px solid ${Colors.lightGray};
+  border-top-width: 1px;
```
Comment (kbrgl) [inline src/screens/Transactions/styles.ts]: You don't need two lines here, you can just do `border-top: `1px solid ${Colors.lightGray}`

File: src/components/Transactions/styles.ts [hunk 7]
```diff
@@ -8,38 +9,82 @@ export const Row = styled.View`
...
+  background: ${Colors.lightGreen};
+  padding: 9px 9px;
+  border: 2px solid ${Colors.lightGreen};
+`;
+
+export const Body1SemiboldText = styled.Text`
+  font-family: 'manrope-semiBold';
+  font-style: normal;
+  font-size: 14px;
+  line-height: 19px;
```
Comment (kbrgl) [inline src/components/Transactions/styles.ts:75]: Specify line height as a decimal ratio (e.g. 1.1, 1.7). In this case it'll be 1.35 since 19/14 = 1.35.

File: src/components/Transactions/styles.ts [hunk 8]
```diff
@@ -8,38 +9,82 @@ export const Row = styled.View`
...
+`;
+
+export const GreenText = styled.Text`
+  color: ${Colors.alertGreen};
+  font-family: 'manrope-bold';
+  font-style: normal;
+  font-size: 12px;
+  line-height: 14px;
```
Comment (kbrgl) [inline src/components/Transactions/styles.ts:90]: Specify line-height as a decimal ratio

File: src/components/Transactions/styles.ts [hunk 9]
```diff
@@ -8,38 +9,82 @@ export const Row = styled.View`
...
+`;
+
+export const RedText = styled.Text`
+  color: ${Colors.alertRed};
+  font-family: 'manrope-bold';
+  font-style: normal;
+  font-size: 12px;
+  line-height: 14px;
```
Comment (kbrgl) [inline src/components/Transactions/styles.ts:98]: Specify line-height as a decimal ratio

File: src/screens/Transactions/styles.ts [hunk 10]
```diff
@@ -1,32 +1,79 @@
...
+export const TitleContainer = styled.View`
+  padding-top: 20px;
+  padding-bottom: 20px;
+`;
+
+export const Title = styled.Text`
+  font-family: 'manrope-bold';
+  font-style: normal;
+  font-size: 42px;
+  line-height: 57px;
```
Comment (kbrgl) [inline src/screens/Transactions/styles.ts:28]: Specify line-height as a decimal ratio

File: src/screens/Transactions/styles.ts [hunk 11]
```diff
@@ -1,32 +1,79 @@
...
+  align-items: flex-start;
+  margin-left: 15%;
+  margin-bottom: 13px;
+`;
+
+export const MediumText = styled.Text`
+  font-family: 'manrope-regular';
+  font-style: normal;
+  font-size: 16px;
+  line-height: 22px;
```
Comment (kbrgl) [inline src/screens/Transactions/styles.ts:66]: Specify line-height as a decimal ratio

File: src/screens/Transactions/styles.ts [hunk 12]
```diff
@@ -1,32 +1,79 @@
...
+  text-align: center;
+`;
+
+export const Body1BoldText = styled.Text`
+  font-family: 'manrope-bold';
+  font-style: normal;
+  font-size: 14px;
+  line-height: 19px;
```
Comment (kbrgl) [inline src/screens/Transactions/styles.ts:74]: Specify line-height as a decimal ratio

File: src/screens/Transactions/styles.ts [hunk 13]
```diff
@@ -1,32 +1,79 @@
 import styled from 'styled-components/native';
+import Colors from '../../../assets/Colors';
+
+export const LogoContainer = styled.View`
+  width: 100%;
+  justify-content: flex-start;
+  margin-left: 15%;
```
Comment (kbrgl) [inline src/screens/Transactions/styles.ts]: Get rid of %age here, replace with pixels

File: src/screens/Transactions/styles.ts [hunk 14]
```diff
@@ -1,32 +1,79 @@
...
 
-export const HorizontalSpacingContainer = styled.View`
-  padding-left: 15px;
-  padding-right: 15px;
+export const BackButtonContainer = styled.View`
+  width: 100%;
+  justify-content: flex-start;
+  align-items: flex-start;
+  margin-left: 10%;
+  margin-top: 15%;
```
Comment (kbrgl) [inline src/screens/Transactions/styles.ts]: Get rid of %age here, replace with pixels

File: src/screens/Transactions/styles.ts [hunk 15]
```diff
@@ -1,32 +1,79 @@
...
+export const LeftAlignedContainer = styled.View`
   width: 100%;
-  flex-direction: row;
-  padding-top: 40px;
-  justify-content: flex-end;
-  margin-top: 20px;
-  margin-right: 60px;
+  justify-content: flex-start;
+  align-items: flex-start;
+  margin-left: 15%;
```
Comment (kbrgl) [inline src/screens/Transactions/styles.ts]: Get rid of %age here, replace with pixels

File: src/components/Transactions/styles.ts [hunk 16]
```diff
@@ -8,38 +9,82 @@ export const Row = styled.View`
...
+  border: 2px solid ${Colors.alertLightRed};
+`;
+
+export const StatusComponentGreen = styled.View`
+  text-align: center;
+  align-items: center;
+  border-radius: 9px;
+  width: 63px;
+  background: ${Colors.lightGreen};
+  padding: 9px 9px;
```
Comment (kbrgl) [inline src/components/Transactions/styles.ts]: Don't need to do `padding: 9px 9px`, can just do `padding: 9px` to specify all at once

File: src/components/Transactions/styles.ts [hunk 17]
```diff
@@ -8,38 +9,82 @@ export const Row = styled.View`
   flex-direction: row;
   align-items: center;
   justify-content: flex-start;
-  border-top: 1px solid black;
-  border-bottom: 1px solid black;
+  height: 67px;
+  border: 0px solid ${Colors.lightGray};
+  border-bottom-width: 1px;
 `;
```
Comment (kbrgl) [inline src/components/Transactions/styles.ts]: Replace with `border-bottom: 1px solid ${Colors.lightGray};`

File: src/screens/Transactions/TransactionsScreen.tsx [hunk 18]
```diff
@@ -1,61 +1,74 @@
...
+}: TransactionStackScreenProps<'TransactionsScreen'>) {
+  const [transactions, setTransactions] = useState<Transaction[]>([]);
+  const { user } = useAuthContext();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        // todo: retrieve vendor uuid
+        if (user) {
+          const vendor = await getVendorByEmail(user.email);
```
Comment (kbrgl) [inline src/screens/Transactions/TransactionsScreen.tsx]: The vendor UUID should probably also be stored in the auth context, so that it only needs to be fetched once.

**Review by kbrgl — CHANGES_REQUESTED** [review]
Great PR! Left some nitpicks in the comments, but overall solid work :)

File: src/components/Transactions/styles.ts (hunk 3 above)
Comment (selene-huang) [reply src/components/Transactions/styles.ts:101]: we'll be having a major refactoring PR next, so I will keep that in mind for then

Comment (sauhardjain) [reply src/components/Transactions/styles.ts:101]: good catch @kbrgl

File: src/screens/Transactions/styles.ts [hunk 19]
```diff
@@ -1,32 +1,79 @@
...
 `;
 
-export const HorizontalSpacingContainer = styled.View`
-  padding-left: 15px;
-  padding-right: 15px;
+export const BackButtonContainer = styled.View`
+  width: 100%;
+  justify-content: flex-start;
+  align-items: flex-start;
+  margin-left: 10%;
```
Comment (sauhardjain) [inline src/screens/Transactions/styles.ts:50]: Use px for margins

**Review by sauhardjain — APPROVED** [review]
GREAT WORK @selene-huang! Everything looks great, some styling nits but we'll go over these when refactoring the flow's styling later.

**Review by kbrgl — CHANGES_REQUESTED** [review]
Looks great! Just one last thing: the LoadingSpinner component should be used as `<LoadingSpinner />`, not as `LoadingSpinner()`. The latter technically works, but it is incorrect. You can see https://github.com/calblueprint/vouchers4veggies/blob/1ca79100b412d6fda0a4a66c68b073bf400d7533/src/navigation/AppNavigator.tsx#L13 for an example.

**Comment by kbrgl** [comment]
> Looks great! Just one last thing: the LoadingSpinner component should be used as `<LoadingSpinner />`, not as `LoadingSpinner()`. The latter technically works, but it is incorrect. You can see
> 
> https://github.com/calblueprint/vouchers4veggies/blob/1ca79100b412d6fda0a4a66c68b073bf400d7533/src/navigation/AppNavigator.tsx#L13
> 
> for an example.

For details on why it is incorrect to *call* a function component, instead of rendering it (as `<LoadingSpinner />`), see here: https://kentcdodds.com/blog/dont-call-a-react-function-component. It can cause hard-to-debug errors!

**Review by kbrgl — APPROVED** [review]
